core/sender.go

```diff
@@ -3,19 +3,41 @@ package core
 import (
 	"github.com/lovelaced/glitzz/logging"
 	"github.com/thoj/go-ircevent"
+	"time"
 )
 
+const messageDelay = 2200 * time.Millisecond
+const messageQueueLength = 100
+
 func NewSender() Sender {
-	return &sender{
-		log: logging.New("core/sender"),
+	rv := &sender{
+		log:              logging.New("core/sender"),
+		outgoingMessages: make(chan outgoingMessage, messageQueueLength),
 	}
+	go rv.run()
```

Comment:
I know that this leaks but we don't have and never will have a scenario when a sender is created and discarded.